server/servers/api/src/main/scala/com/prisma/api/schema/ObjectTypeBuilder.scala

```diff
@@ -261,131 +254,50 @@ class ObjectTypeBuilder(
-              // todo: handle this case
-            }
-            None
-
-          case (Some(value), true) =>
-            def mapTo[T](value: Any, convert: JsValue => T): Seq[T] = {
-              value match {
-                case x: String =>
-                  Try {
-                    x.parseJson.asInstanceOf[JsArray].elements.map(convert)
-                  } match {
-                    case Success(x) => x
-                    case Failure(e) => e.printStackTrace(); Vector.empty
-                  }
-
-                case x: Vector[_] =>
-                  x.map(_.asInstanceOf[T])
-              }
-            }
-
-            field.typeIdentifier match {
-              case TypeIdentifier.String    => mapTo(value, x => x.convertTo[String])
-              case TypeIdentifier.Int       => mapTo(value, x => x.convertTo[Int])
-              case TypeIdentifier.Float     => mapTo(value, x => x.convertTo[Double])
-              case TypeIdentifier.Boolean   => mapTo(value, x => x.convertTo[Boolean])
-              case TypeIdentifier.GraphQLID => mapTo(value, x => x.convertTo[String])
-
-              case TypeIdentifier.DateTime => mapTo(value, x => new DateTime(x.convertTo[String], DateTimeZone.UTC))
-              case TypeIdentifier.Enum     => mapTo(value, x => x.convertTo[String])
-              case TypeIdentifier.Json     => mapTo(value, x => x.convertTo[JsValue])
-            }
-
-          case (Some(value), false) =>
-            def mapTo[T](value: Any) = value.asInstanceOf[T]
-
-            field.typeIdentifier match {
-              case TypeIdentifier.String    => mapTo[String](value)
-              case TypeIdentifier.Int       => mapTo[Int](value)
-              case TypeIdentifier.Float     => mapTo[Double](value)
-              case TypeIdentifier.Boolean   => mapTo[Boolean](value)
-              case TypeIdentifier.GraphQLID => mapTo[String](value)
-              case TypeIdentifier.DateTime =>
-                value.isInstanceOf[DateTime] match {
-                  case true => value
-                  case false =>
-                    value.isInstanceOf[java.sql.Timestamp] match {
-                      case true =>
-                        DateTime.parse(value.asInstanceOf[java.sql.Timestamp].toString,
-                                       DateTimeFormat
-                                         .forPattern("yyyy-MM-dd HH:mm:ss.SSS")
-                                         .withZoneUTC())
-                      case false => new DateTime(value.asInstanceOf[String], DateTimeZone.UTC)
-                    }
-                }
-              case TypeIdentifier.Enum => mapTo[String](value)
-              case TypeIdentifier.Json => mapTo[JsValue](value)
-            }
-        }
+  def convertScalarFieldValueFromDatabase(field: models.Field, item: PrismaNode): Any = {
+    import com.prisma.util.json.PlaySprayConversions._
+
+    //todo use GCValueextractor once we got rid of Spray
```

Comment:
Do you have an issue to remove Spray where we track things we can clean up then?

Reply:
Opened One